File: 119.pascals-triangle-ii.js
/*
 * @lc app=leetcode id=119 lang=javascript
 *
 * [119] Pascal's Triangle II
 */

// @lc code=start
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function (rowIndex) {
    if (rowIndex === 0) return [1]
    if (rowIndex === 1) return [1, 1]
    const triangle = [[1], [1, 1]]
    let numRows = rowIndex

    while (numRows !== 1) {
        const last = triangle[triangle.length - 1]
        const temp = [1]
        for (let i = 0; i < last.length; i++) {
            temp.push(last[i] + last[i + 1])
        }
        temp.push(1)
        triangle.push(temp)
        numRows--
    }
    return triangle[rowIndex - 1]
};
// @lc code=end

File: 119.pascals-triangle-ii.py
#
# @lc app=leetcode id=119 lang=python
#
# [119] Pascal's Triangle II
#

# @lc code=start
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        if rowIndex == 0:
            return [1]
        elif rowIndex == 1:
            return [1, 1]
        row = [1, 1]
        for i in range(2, rowIndex + 1):
            row = [1] + [row[j] + row[j+1] for j in range(i-1)] + [1]
        return row

# @lc code=end
